specs/safe-liveness-checking.md

```diff
@@ -0,0 +1,70 @@
+# Safe Liveness Checking
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Liveness Checking Mechanism](#liveness-checking-mechanism)
+- [Liveness checking methodology](#liveness-checking-methodology)
+  - [The Liveness Guard](#the-liveness-guard)
+  - [The Liveness Module](#the-liveness-module)
+  - [Shutdown](#shutdown)
+  - [Security Properties](#security-properties)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
```

Comment:
A mermaid diagram could be very helpful here showing the call flow, chatgpt is good at generating them if you just tell it what to do and iterate on it through multiple prompts

Reply:
done